test/basics.test.js

```diff
@@ -70,4 +70,23 @@ test('basics', () => {
 		).toBe(undefined);
 		expect(callback).toBeCalledWith(undefined);
 	});
+	// apply with object, which has a then getter which throws.
+	run(createCallback(), callback => {
+		const throwingGetter = {
+			get then() { throw new Error() }
+		}
+		expect(
+			() => apply(throwingGetter, callback)
+		).not.toThrow();
+		expect(callback).toHaveBeenCalledWith(throwingGetter);
+	});
+	// apply with object, where any getter throws.
+	run(createCallback(), callback => {
```

Comment:
I couldn't test those in one go, as it turns out using `expect(callback)` with the proxy that always fails, calls the getters and thus breaks. Likely this is a jest feature getting info on the passed args.

Reply:
`expect` blows up with the hostile proxy as well, I see!

I worked around this by making _just_ `then` throw.